### ISSUE TITLE
Yet another Layout?

### ISSUE THREAD

**Issue by TheJP** [RoboDurden/Hoverboard-Firmware-Hack-Gen2.x#28]
Hello :)

Just found this repo and not sure if my boards resemble any of the already provided layouts.

Could someone maybe point out which of the two white connectors is meant for the on/off button? The other should be an LED port, but confusingly uses the exact same connector. (I'm a novice in hardware but an experienced programmer.)

# Probably Master (Side where the on/off button is located):

Chip is a: GD32F103

![IMG_20230928_172044](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/5956390/bca35870-ee8d-473f-9d15-fb63a2b7e7da)

# Probably Slave (Other Side)

![IMG_20230927_135342](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/5956390/75253435-6f80-4855-b1f3-0a446998730f)

**Comment by RoboDurden** [comment]
The onoff button is probably to the right where the dcdc step down converter for the 5V and 3.3V is generated.
The first board with the buzzer is probably the slave and the second one with the dcdc step down components is the master.
So there i guess you should plug in the onoff button.

Please make a closer photo of the 48 pin MCU so we can see if it is a GD23F130 or some other chip.

**Comment by TheJP** [comment]
Both boards seem to be nearly identical (I didn't find a difference so far). I guessed the top picture to be the master, because the on/off button is located on that side of the hoverboard.

> The onoff button is probably to the right where the dcdc step down converter for the 5V and 3.3V is generated.

Thanks for the hint. So the top right next to the buzzer would be the on/off plug?
It has to be one of those two-pin plugs (top or bottom of the board, just unsure which one), And the on/off has to be connected to the board on the first image.

> Please make a closer photo of the 48 pin MCU so we can see if it is a GD23F130 or some other chip.

It reads GD32F130 directly on the MCU. I tried creating a better picture, but my phone cannot focus when I go any closer than the following. But I think on that picture it is kind of readable. (Both boards have the same MCU, this is the board from my first picture.)

![IMG_20230928_190015](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/5956390/fdb5d67b-4e8c-407a-9f4d-47ae0fd79e8f)

**Comment by RoboDurden** [comment]
Sorry i rotated the photos: https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/tree/main/Schematics_2.12

This is the dcdc hardware of the master (without the buzzer) and probably the onoff switch:
![grafik](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/12238841/40066bf8-107b-4704-9860-3c677364bf43)

**Comment by RoboDurden** [comment]
Yes, a GD32F130C8 is compatible with this firmware and the simpleFOC firmware will also run on it.
Now you need to trace the pins.
hall sensor
onoff button
hold button
led buttons

Please make a high resolution photo of the backside and the fronside (with the cables pushed out of the way).

**Comment by RoboDurden** [comment]
A sorry, this is no coil but simply the tranistor bent to the left :-(
Both boards have a dcdc step down:
![grafik](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/12238841/85d5ca90-5d0d-4857-b79f-811b8e06e6c9)

So the board with the buzzer is the master and the onoff button should be to the left.

**Comment by TheJP** [comment]
Both boards have a buzzer and both are directly connected to the power-source.

> So the board with the buzzer is the master and the onoff button should be to the left.

Is left near the buzzer or far from the buzzer?

**Comment by RoboDurden** [comment]
when you disassemble everything make new photos.
hall sensor header is down, buzzer left, mosfets top. onoff persumably right.

**Comment by TheJP** [comment]
Backside

![IMG_20230928_192337](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/5956390/edddcfae-4c55-4782-8d77-a353eb22310b)

**Comment by RoboDurden** [comment]
and a nice photo of the frontside with cables pushed out of sight.

**Comment by TheJP** [comment]
Wasn't sure which cables from the front were bothering you. So here are pictures where I temporarily disconnected the ports that lead to the other board. (Same board twice, 1. without flash, 2. with flash)

![IMG_20230928_193145](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/5956390/4dd57993-1420-46c6-aa62-4542329b856e)

![IMG_20230928_193156](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/5956390/b4cf2bf8-ed69-4373-af9e-d4dfc9961aa5)

**Comment by RoboDurden** [comment]
Okay, the mosfet pins mostly are the same for all gd32F130 boards.
And the hall pins seem to be PA0, PA1 and PA2
![grafik](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/12238841/8b635d5c-301c-476b-b735-8d210fcc857d)

The 5 led pins most likely are PA15, PB3, PB4, PB5 and PB6

Now you need a cheap st-link-v2 dongle to unlock the boot loader and install the Keil ide to compile and test a new defines_2-12.h

And a constant current max 2A power supply (or step down converter with CC).

**Comment by RoboDurden** [comment]
no i am not sure about the led.
But this enough for me for now.
good night from Germany

**Comment by TheJP** [comment]
Thanks for looking into it! :)

I have an st-link-v2 USB dongle. For a power supply I have an original hoverboard li-ion battery pack.
Probably don't have time to test it out until next week. I'll write here for an update.

Good night from Switzerland

**Comment by Hishamahmed97** [comment]
@TheJP @RoboDurden Hello guys,
Can you help how I can correctly build the keil project for this GD32F103 ?

**Comment by RoboDurden** [comment]
And for sure do not post triple times :-((
